receiver/apachesparkreceiver/metadata.yaml

```diff
@@ -4,4 +4,674 @@ status:
   class: receiver
   stability: 
     development: [metrics]
-  distributions: [contrib]
\ No newline at end of file
+  distributions: [contrib]
+
+resource_attributes:
+  spark.application.id:
+    description: The ID of the application for which the metric was recorded.
+    type: string
+    enabled: true
+  spark.application.name:
+    description: The name of the application for which the metric was recorded.
+    type: string
+    enabled: true
+  spark.stage.id:
+    description: The ID of the application stage for which the metric was recorded.
+    type: int
+    enabled: true
+  spark.stage.attempt.id:
+    description: The ID of the stage attempt for which the metric was recorded.
+    type: int
+  spark.executor.id:
+    description: The ID of the executor for which the metric was recorded.
+    type: string
+    enabled: true
+  spark.job.id:
+    description: The ID of the job for which the metric was recorded.
+    type: int
+    enabled: true
+
+attributes:
+  stage_active:
+    description: Whether the stage for which the metric was recorded is active.
+    type: bool
+  stage_complete:
+    description: Whether the stage for which the metric was recorded is complete.
+    type: bool
+  stage_pending:
+    description: Whether the stage for which the metric was recorded is pending.
+    type: bool
+  stage_failed:
+    description: Whether the stage for which the metric was recorded is failed.
+    type: bool
+  stage_task_result:
+    description: The result of the stage tasks for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+      - killed
+  executor_task_result:
+    description: The result of the executor tasks for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+  job_task_result:
+    description: The result of the job tasks for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+      - skipped
+  job_stage_result:
+    description: The result of the job stages for which the metric was recorded.
+    type: string
+    enum:
+      - completed
+      - failed
+      - skipped
```

Comment:
For all of these, let's use the `name_override` option to shorten the names. Basically, we need the keys to be unique in this file, but we don't need all the context when attaching them to metrics (e.g. `stage_task_result` can just be called `result` when attached to a stage.task.result metric.)